optimum/amd/brevitas/quantizer.py

```diff
@@ -0,0 +1,327 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+# Licensed under the MIT License.
+
+import inspect
+import logging
+from typing import Dict, List, Optional, Union
+
+import torch
+from brevitas.graph.calibrate import calibration_mode, bias_correction_mode
+from brevitas.graph.equalize import activation_equalization_mode
+from brevitas.graph.gptq import gptq_mode
+from brevitas_examples.common.generative.quantize import quantize_model
+from brevitas_examples.llm.llm_quant.equalize import apply_weight_equalization
+from brevitas_examples.llm.llm_quant.prepare_for_quantize import replace_mha_with_quantizable_layers
+from tqdm import tqdm
+
+from optimum.exporters import TasksManager
+from optimum.quantization_base import OptimumQuantizer
+from transformers.utils.fx import symbolic_trace
+
+from .accelerate_utils import offload_model, remove_hooks
+from .configuration import BrevitasQuantizationConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+class BrevitasQuantizer(OptimumQuantizer):
+    """
+    Handles the Runtime quantization process for models shared on huggingface.co/models.
+    """
+
+    def __init__(self, model: torch.nn.Module):
+        super().__init__()
+        self.model = model
+        self.config = self.model.config
+        self.group_of_parallel_layers = None
+
+    def validate_quant_config(self, quantization_config: BrevitasQuantizationConfig):
+        dtype = next(iter(self.model.parameters()))
+        if dtype == torch.bfloat16 and quantization_config.replace_mha_with_quantizable:
```

Comment:
Shall we list all the invalid combinations?
Some of them will fail at runtime, e.g., dynamic per group asymmetric activation quantization is not supported.

Reply:
Actually, I noticed that the post_init of the config does also some checks, so maybe that type of should live there